internal/store/driver.go

```diff
@@ -251,8 +264,9 @@ func (d *Driver) Sync(ctx context.Context, c client.Client) error {
 		for _, currTunnel := range currTunnels.Items {
 			if desiredTunnel.Name == currTunnel.Name && desiredTunnel.Namespace == currTunnel.Namespace {
 				// It matches so lets update it if anything is different
-				if !reflect.DeepEqual(desiredTunnel.Spec, currTunnel.Spec) {
+				if !reflect.DeepEqual(desiredTunnel.Spec, currTunnel.Spec) || !hasControllerManagedAnnotation(&currTunnel) {
```

Comment:
always curious when I see DeMorgan situations like this if `!(A & B)` vs `!A || !B` feels more readable.

Reply:
We historically haven't adopted a style guides around these types of preferences, if we feel strongly enough I would recommend spinning up a conversation or doc outside of this PR. 

I very much appreciate anyone who knows DeMorgan's law and formal logic in general 😄